src/main/ruby/truffleruby/core/rational.rb

```diff
@@ -342,25 +346,94 @@ def self.convert(num, den)
     when Integer
       # nothing
     when String
-      num = String::Rationalizer.new(num).strict_convert
+      if exception
+        num = String::Rationalizer.new(num).strict_convert
+      else
+        num = String::Rationalizer.new(num).convert(nil)
+        return nil if Primitive.nil?(num)
+      end
     when Float, Complex
       num = num.to_r
+    else
+      if !num.respond_to?(:to_r)
+        begin
+          integer = Truffle::Type.rb_check_to_integer(num, :to_int)
+          num = integer if integer
+        rescue Exception
+        end
+      end
```

Comment:
`rescue Exception` is imitating `rb_protected`. Wondering if there is a better way.